bundles/org.eclipse.ui.ide/src/org/eclipse/ui/internal/ide/ChooseWorkspaceDialog.java

```diff
@@ -464,6 +471,14 @@ protected Combo createPathCombo(Composite panel) {
 					}
 				}
 				okButton.setEnabled(nonWhitespaceFound);
+				try {
+				File location = new File(characters);
+				label.setText(
```

Comment:
The label should only be visible (ie not empty) if the input is different from the resolved normalized location. When the path is absolute and doesn't contain `.` or `..` or whatever (basically when the resolved location is equal to the input) it would be better to not display it/make the label empty.

Reply:
given the main use case is to show the user  what the path would resolve to , how this will work for `~` character?
for example, if the path contains `~`, then the absolute normalised path will make no change and hence both will be the same. In this scenario the label should show up to warn the user, while it will not show up according to your suggestion?

Reply:
OK, I understand the purpose.
If `~` is an exceptional case, we can maybe just show a warning, setting the label to something like `⚠️ '~' is not expanded, resolved path is ...`?

Reply:
@mickaelistria I followed your suggestion, however I have difficulty in rendering the image. with the pushed changes, I am getting NPE. I know the reason as the variable is uninitialised but the same logic is working elsewhere in the project. Can you advise?

`java.lang.NullPointerException: Cannot invoke "org.eclipse.e4.ui.workbench.IResourceUtilities.imageDescriptorFromURI(org.eclipse.emf.common.util.URI)" because "this.resUtils" is null
`

Reply:
Don't both with the image, just use the warning emoji in messages.properties/ChooseWorkspaceDialog_TildeNonExpandedWarning0 (IIRC you should use the escaped encoding of the emoji in properties file, eg `\u26A0\uFE0F`

Reply:
@mickaelistria I have addressed the requested changes. Please have a look.